common/Environments/Models/CardProperty.cs

```diff
@@ -83,33 +111,39 @@ public void UpdateTitleBasedOnPropertyKind(string title, ComputeSystemPropertyKi
     /// Converts a passed in ms-resource URI and package full name to a BitmapImage.
     /// </summary>
     /// <param name="iconPathUri">the ms-resource:// path to an image resource in an app packages pri file.</param>
-    /// <param name="packageFullName">The package full name where the resource is located.</param>
     /// <returns>The bitmap image that represents the icon.</returns>
-    public unsafe BitmapImage ConvertMsResourceToIcon(Uri iconPathUri, string packageFullName)
+    public static unsafe BitmapImage ConvertMsResourceToIcon(Uri iconPathUri, string packageFullName)
     {
-        var indirectPathToResource = "@{" + packageFullName + "?" + iconPathUri.AbsolutePath + "}";
-        Span<char> outputBuffer = new char[512];
-
-        fixed (char* outBufferPointer = outputBuffer)
+        try
         {
-            fixed (char* resourcePathPointer = indirectPathToResource)
+            var indirectPathToResource = "@{" + packageFullName + "? " + iconPathUri.AbsoluteUri + "}";
+            Span<char> outputBuffer = new char[512];
```

Comment:
512 characters feels too small these days. It probably wouldn't hurt to allocate 1K.